crates/starknet/src/transactions/declare_transaction_v2.rs

```diff
@@ -70,43 +102,94 @@ impl DeclareTransactionV2 {
 
 impl HashProducer for DeclareTransactionV2 {
     fn generate_hash(&self) -> DevnetResult<Felt> {
-        let class_hash = self.class_hash.unwrap_or(self.sierra_contract_class.generate_hash()?);
-
-        let calldata = [class_hash.into()].to_vec();
-        let additional_data = [self.nonce.into(), self.compiled_class_hash.into()].to_vec();
-
-        let transaction_hash: Felt = calculate_transaction_hash_common(
-            TransactionHashPrefix::Declare,
-            self.version.into(),
-            &self.sender_address.try_into()?,
-            Felt::from(0).into(),
-            &calldata,
-            self.max_fee,
-            self.chain_id.into(),
-            &additional_data,
-        )
-        .map_err(|err| {
-            starknet_types::error::Error::TransactionError(
-                starknet_in_rust::transaction::error::TransactionError::Syscall(err),
-            )
-        })?
-        .into();
-
-        Ok(transaction_hash)
+        Ok(self.inner.hash_value.clone().into())
     }
 }
 
 #[cfg(test)]
 mod tests {
 
+    use serde::Deserialize;
+    use starknet_in_rust::core::contract_address::compute_sierra_class_hash;
+    use starknet_in_rust::definitions::block_context::StarknetChainId;
+    use starknet_rs_core::types::contract::SierraClass;
+    use starknet_types::contract_address::ContractAddress;
+    use starknet_types::contract_class::ContractClass;
+    use starknet_types::felt::Felt;
+    use starknet_types::traits::{HashProducer, ToHexString};
+
+    use super::DeclareTransactionV2;
     use crate::utils::test_utils::{
         dummy_cairo_1_contract_class, dummy_contract_address, dummy_felt,
+        get_transaction_from_feeder_gateway,
     };
 
-    #[ignore]
```

Comment:
I see there is still one test where `#[ignore]` is left. What about that?

Reply:
yes

Reply:
A check will be added in this PR

Reply:
A check will be added in this PR

Reply:
[5f89dc8](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/122/commits/5f89dc86194a22c8d6cd2dbe53870ee20f1cbfbb)